packages/components/src/primitives/svg/index.native.js

```diff
@@ -1,13 +1,33 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
 import {
-	Path as PATH,
+	Path,
+	G,
 	Svg,
 } from 'react-native-svg';
 
-export const Path = PATH;
+export {
+	G,
+	Path,
+};
+
 export const SVG = ( props ) => {
+	// We're using the react-native-classname-to-style plugin, so when a `className` prop is passed it gets converted to `style` here.
+	// Given it carries a string (as it was originally className) but an object is expected for `style`,
+	// we need to check whether `style` exists and is a string, and strip in that case.
+	const safeProps = ( typeof props.style === 'string' || props.style instanceof String ) ? { ...omit( props, [ 'style', 'className' ] ) } : { ...omit( props, [ 'className' ] ) };
```

Comment:
Feels super weird to have to manually strip the style definition so, if you don't mind @mzorz , let's dig a little deeper in this one.

We already had a chat over Slack with what probably needs to be done here (properly load the styles for RN) so, let's wait a bit to finish that effort, is that OK?

Reply:
As discussed with @hypest , made the `style` convert from string to object properly (if it applies) in 2d474511dd100604421564ea3cfb745c300f7cc7 and b678c6d2abba31a627dd5eb9d307cb36c49aede1. 

To test, I added the following:
```
.dashicons-editor-bold {
	transform: rotate(20deg);
}

```
to `packages/components/src/dashicon/style.scss` 

and the styles get passed to the svg component, this is how the bold icon gets rendered:

<img width="307" alt="screen shot 2018-09-06 at 13 21 46" src="https://user-images.githubusercontent.com/6597771/45171725-860fc480-b1d9-11e8-9fac-1c0bba64c3f3.png">

See the **B** icon slightly leaning to the right? 😄

Reply:
Out of curiosity, why does it happen that style becomes a string?
On the web you have to pass an object as style, I think this is also the case in React Native. So this is very surprising. I would love to understand why this is something we need to handle in the first place.

Reply:
As far as I understand, `className` is [not supported in React Native](https://github.com/facebook/react-native/issues/7238), so we need to define that as a prop on each object we'd want to use that in.
Because of this and being that `className` is being used in different places in Gutenberg, we're using a [Babel plugin](https://github.com/kristerkari/babel-plugin-react-native-classname-to-style) that transforms `className` to `style` in RN so to make it transparent to Gutenberg code.

AFAIU the plugin just keeps the passed object as it was without making any other conversion, so if the original object was a string, it will be a string after going through it as well.

So knowing this, it looks like the problem arises then because Dashicon `gutenberg/packages/components/src/dashicon/index.js` declares `className={ iconClass }` where `iconClass` is a string, a defined here: https://github.com/WordPress/gutenberg/blob/master/packages/components/src/dashicon/index.js#L899

While `className` can be a string (as it originally is in Dashicon code), `style` should be an object instead, but we're not transforming it (just keeping the same type).

Does that make sense? Should we do something differently? cc @hypest as well to see if there's something else to add / do differently in this regard.